itests/fevm_test.go

```diff
@@ -119,22 +275,325 @@ func TestFEVMDelegateCall(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, result, expectedResultActor)
 }
 
+// TestFEVMDelegateCallRevert makes a delegatecall action and then calls revert.
+// the state should not have changed
+func TestFEVMDelegateCallRevert(t *testing.T) {
+	ctx, cancel, client := kit.SetupFEVMTest(t)
+	defer cancel()
+
+	//install contract Actor
+	filenameActor := "contracts/DelegatecallActor.hex"
+	fromAddr, actorAddr := client.EVM().DeployContractFromFilename(ctx, filenameActor)
+	//install contract Storage
+	filenameStorage := "contracts/DelegatecallStorage.hex"
+	fromAddrStorage, storageAddr := client.EVM().DeployContractFromFilename(ctx, filenameStorage)
+	require.Equal(t, fromAddr, fromAddrStorage)
+
+	//call Contract Storage which makes a delegatecall to contract Actor
+	//this contract call sets the "counter" variable to 7, from default value 0
+
+	inputDataContract := inputDataFromFrom(ctx, t, client, actorAddr)
+	inputDataValue := inputDataFromArray([]byte{7})
+	inputData := append(inputDataContract, inputDataValue...)
+
+	//verify that the returned value of the call to setvars is 7
+	_, _, err := client.EVM().InvokeContractByFuncName(ctx, fromAddr, storageAddr, "setVarsRevert(address,uint256)", inputData)
+	require.Error(t, err)
+	t.Log(err)
+
+	//test the value is 0 via calling the getter and was not set to 7
+	expectedResult, err := hex.DecodeString("0000000000000000000000000000000000000000000000000000000000000000")
+	require.NoError(t, err)
+	result, _, err := client.EVM().InvokeContractByFuncName(ctx, fromAddr, storageAddr, "getCounter()", []byte{})
+	require.NoError(t, err)
+	require.Equal(t, result, expectedResult)
+
+	//test the value is 0 via calling the getter on the Actor contract
+	result, _, err = client.EVM().InvokeContractByFuncName(ctx, fromAddr, actorAddr, "getCounter()", []byte{})
+	require.NoError(t, err)
+	require.Equal(t, result, expectedResult)
+}
+
+// TestFEVMDelegateCallRevert makes a call that is a simple revert
+func TestFEVMSimpleRevert(t *testing.T) {
+	ctx, cancel, client := kit.SetupFEVMTest(t)
+	defer cancel()
+
+	//install contract Actor
+	filenameStorage := "contracts/DelegatecallStorage.hex"
+	fromAddr, contractAddr := client.EVM().DeployContractFromFilename(ctx, filenameStorage)
+
+	//call revert
+	_, _, err := client.EVM().InvokeContractByFuncName(ctx, fromAddr, contractAddr, "revert()", []byte{})
+
+	require.Error(t, err)
+	t.Log(err)
+}
+
+// TestFEVMSelfDestruct creates a contract that just has a self destruct feature and calls it
+func TestFEVMSelfDestruct(t *testing.T) {
+	ctx, cancel, client := kit.SetupFEVMTest(t)
+	defer cancel()
+
+	//install contract Actor
+	filenameStorage := "contracts/SelfDestruct.hex"
+	fromAddr, contractAddr := client.EVM().DeployContractFromFilename(ctx, filenameStorage)
+
+	//call destroy
+	_, _, err := client.EVM().InvokeContractByFuncName(ctx, fromAddr, contractAddr, "destroy()", []byte{})
+	t.Log(err)
```

Comment:
I don't think you need t.Log(err) in places where you're asserting no error. The test will print the error if there is one.

Reply:
good point! Definitely was in a few rabbit holes around debugging while coding this and peppered a lot of logging we don't need anymore

Reply:
I removed a lot of unhelpful t.Log calls